orcestra/postprocess/level0.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+
+def _bahamas_fix_time(ds):
+    """Fix time coordinate of BAHAMAS datasets."""
+    return ds.rename({"tid": "time", "TIME": "time"}).set_index(time="time")
+
+
+def _bahamas_line_of_sight(ds):
+    """Calculate the length of line-of-sight (LOS)."""
+    return ds.assign(
+        LOS=lambda dx: dx.IRS_ALT
+        / np.cos(np.radians(dx.IRS_THE))
+        / np.cos(np.radians(dx.IRS_PHI))
+    )
+
+
+def bahamas(ds):
+    """Post-processing of BAHAMAS datasets."""
+    return ds.pipe(
+        _bahamas_fix_time,
+    ).pipe(
+        _bahamas_line_of_sight,
+    )
+
+
+def _radar_ql_fix_time(ds):
+    """Fix time coordinate of RADAR QL datasets."""
+    datetime = np.datetime64("1970-01-01", "ns")
+    datetime += ds.time.values * np.timedelta64(1, "s")
+    datetime += ds.microsec.values * np.timedelta64(1, "us")
```

Comment:
I'd generally recommend not to modify variables, but rather to assign them once. In this case, e.g.:
```suggestion
    datetime = np.datetime64("1970-01-01", "ns") + ds.time.values * np.timedelta64(1, "s") + ds.microsec.values * np.timedelta64(1, "us")
```
(maybe in multiple lines)

Reply:
Good catch! I missed that this modifies the `ds.time` array